### PR TITLE
Add pgRouting 3.3.2, 3.2.2, 3.1.4, 3.0.6, postgresql-14 and upgrade osm2pgrouting to 2.3.8

### PR DESCRIPTION
**Pull request by krashish8**
Changes proposed in this pull request:
- Upgraded osm2pgrouting to 2.3.8.
- Changed master -> main.
- Updated all main and develop branch images, and added postgresql-14
- Add pgrouting 3.0.6, 3.1.4, 3.2.2, 3.3.2 each with pg11, 12, 13, 14 and postgis-3.2

@pgRouting/admins

### CODE REVIEW

**Comment by krashish8** [comment]
I have some queries, as I'm updating these Docker images for the first time:

1. There can be a lot of combinations possible with different pgrouting, postgis, and PostgreSQL versions. Do we update for all such combinations? (I saw that we update only the latest ones, e.g. pgrouting 3.1.3 is present but 3.1.2 is absent)
    - So, I added only the latest docker images for pgrouting 3.0, 3.1, 3.2, and 3.3, skipping any old patch releases.
    - Added them for all the 4 PostgreSQL versions - 11, 12, 13, 14.
    - Only added them for the latest PostGIS v3.2

2. Do we check whether the docker images are correct or not? If yes, how? I don't see any CI builds/checks present here.
    - We should check just to be sure that, say, a PostGIS version 3.2 is compatible with pgrouting 3.0.6 and with pg14.

3. I have renamed master to main, and updated the osm2pgrouting version. So, in Dockerhub: all the Docker images need to be updated (because of osm2pgrouting) and the Docker images named "master" need to be deleted, right? (to create the new images named "main")

Please review, and do suggest if some changes are required.

**Comment by cvvergara** [comment]
In https://github.com/pgRouting/docker-pgrouting/pull/49/commits/b9021c6ab96dc8e0e1f22cdd99ee5de452fbdede

I see that you are also changing version 2.x on osm2pgrouting.

We are no longer supporting 2.x
I think: do not encourage users to use 2.x

File: 10-2.5-master/README.md [hunk 1]
```diff
@@ -1,3 +0,0 @@
-# pgRouting master (pg10)
```
Comment (cayetanobv) [inline 10-2.5-master/README.md:1]: Why did you remove this README file?

File: 11-2.5-master/README.md [hunk 2]
```diff
@@ -1,3 +0,0 @@
-# pgRouting master (pg11)
```
Comment (cayetanobv) [inline 11-2.5-master/README.md:1]: Why did you remove this README file?

**Review by cayetanobv — CHANGES_REQUESTED** [review]
Several minor comments. Well done. Thanks for your work.

File: 10-2.5-develop/extra/Dockerfile [hunk 3]
```diff
@@ -1,6 +1,6 @@
 FROM pgrouting/pgrouting:10-2.5-develop
 
-ENV OSM2PGROUTING_VERSION 2.3.7
+ENV OSM2PGROUTING_VERSION 2.3.8
```
[the remark this answers is not in the document]
Comment (cayetanobv) [reply 10-2.5-develop/extra/Dockerfile:3]: Yes, you are right, 2.5 is not supported:
https://trac.osgeo.org/postgis/wiki/UsersWikiPostgreSQLPostGIS

And PostgreSQL 10 is close to the final release:
https://www.postgresql.org/support/versioning/

[the remark this answers is not in the document]
Comment (sanak) [reply 10-2.5-develop/extra/Dockerfile:3]: I think that 2.5 is for PostGIS version.
https://github.com/postgis/docker-postgis

[the remark this answers is not in the document]
Comment (sanak) [reply 10-2.5-develop/extra/Dockerfile:3]: Well, docker-postgis still supports PostGIS 2.5 on PostgreSQL 10/11, so if pgRouting `develop` branch supports those PostGIS/PostgreSQL versions, I think that this is no problem.
https://github.com/postgis/docker-postgis#versions--2022-06-22-

**Review by sanak — APPROVED** [review]
LGTM

**Comment by cvvergara** [comment]
marked as draft after #50 is resolved

File: 10-2.5-develop/extra/Dockerfile (hunk 3 above)
[the remark this answers is not in the document]
Comment (krashish8) [reply 10-2.5-develop/extra/Dockerfile:3]: It is Postgresql 10, PostGIS 2.5 and pgRouting develop.
I agree with @sanak, if docker-postgis also supports PostGIS 2.5 on PostgreSQL 10/11, we can also support this.

File: 11-2.5-master/README.md (hunk 2 above)
Comment (krashish8) [reply 11-2.5-master/README.md:1]: Same as above

**Comment by krashish8** [comment]
Q: Why are we having different docker images for different pgRouting patch releases?
Maybe we can have a single image `12-3.1-3.2` containing the latest patch release 3.2.2, instead of several images `12-3.1-3.2.0`, `12-3.1-3.2.1`, etc because anyway, users would want to use the latest patch release.

**Comment by irwin-uc** [comment]
Why can't Docker pull 13-3.1-3.2.1 ?

**Comment by krashish8** [comment]
@irwin-uc This is just a draft PR, probably won't merge now, need to make some changes. Actually, we want to have a single docker image for a minor release rather than several docker images per patch release because anyway, users would want to use the latest patch version. (older patch version might have bugs which is fixed in later patch version)

Refer: https://github.com/pgRouting/docker-pgrouting/issues/53

**Comment by krashish8** [comment]
Closing because of #55

**Pull request closed without merging**